x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/FullTextFunction.java

```
@@ -218,8 +225,14 @@ private static void checkFullTextQueryFunctions(LogicalPlan plan, Failures failu
                 m -> "[" + m.functionName() + "] " + m.functionType(),
                 failures
             );
-            checkFullTextSearchDisjunctions(condition, ftf -> "[" + ftf.functionName() + "] " + ftf.functionType(), failures);
             checkFullTextFunctionsParents(condition, failures);
+
+            boolean usesScore = plan.output()
+                .stream()
+                .anyMatch(attr -> attr instanceof MetadataAttribute ma && ma.name().equals(MetadataAttribute.SCORE));
+            if (usesScore) {
+                checkFullTextSearchDisjunctions(condition, failures);
```

We actually can get scores for some disjunctions if both sides of the `OR` can be pushed down, however it is decided at data nodes, the coordinator node doesn't have this information. 

For example, if the `checkFullTextSearchDisjunctions` is commented out, this query below can get score. 

```
+ curl -u elastic:password -v -X POST 'localhost:9200/_query?format=txt&pretty' -H 'Content-Type: application/json' '-d
{
  "query": "from sample_data metadata _score | where match(message, \"Connected\") or cidr_match(client.ip, \"172.21.3.15\") | drop @timestamp, event.duration, test_date"
}
'
   client.ip   |       message       |      _score      
---------------+---------------------+------------------
172.21.2.162   |Connected to 10.1.0.3|0.7329744100570679
172.21.2.113   |Connected to 10.1.0.2|0.7329744100570679
172.21.3.15    |Connection error     |0.0               
172.21.3.15    |Connection error     |0.0               
172.21.3.15    |Connection error     |0.0               
172.21.3.15    |Connected to 10.1.0.1|0.7329744100570679   
```

I'm wondering if it is a good idea to not fail queries with disjunctions and `metadata _score`? One option that I can think of is that if data nodes can pushdown the `OR` tree to Lucene,  _score is populated by the one returned by Lucene, otherwise if we end up with running full text function with `LuceneQueryExpressionEvaluator`, we can set  _score to null with warnings that mention score is not available for this situation. This is not a new behavior in `ES|QL`, for the other `ES|QL` functions, when there are exceptions happen at data nodes, null + warnings will be returned.

---

> I'm wondering if it is a good idea to not fail queries with disjunctions and metadata _score? 

I wondered about this as well. I think it's better to fail:
- Score would be useless in this situation for relevance scoring
- Users have a workaround: Either disable scoring or use full text functions for the query
- This is a temporary limitation that we're aiming to fix

I'd prefer to fail early rather than provide a broken score, even with warnings.